client/src/js/services/JournalPostingModalService.js

```diff
@@ -0,0 +1,173 @@
+angular.module('bhima.services')
+  .service('JournalPostingModalService', JournalPostingModalService);
+
+/** Dependencies injection */
+JournalPostingModalService.$inject = ['util', '$http', '$uibModal'];
+
+function JournalPostingModalService(util, $http, Modal) {
+  var service = this;
+  var baseUrl = '/trial_balance/';
+
+  service.parseSelectedGridRecord = parseSelectedGridRecord;
+  // service.postingModalService = postingModalService;
+  // service.getCurrentGroupingColumn = getCurrentGroupingColumn;
+  service.switchGroup = switchGroup;
+  service.getDataByAccount = getDataByAccount;
+  service.checkTransactions = checkTransactions;
+  service.getFeedBack = getFeedBack;
+  service.getCSSClass = getCSSClass;
+  service.openErrorViewerModal = openErrorViewerModal;
+  service.postToGeneralLedger = postToGeneralLedger;
+
+  function openErrorViewerModal(errors, feedBack, cssClass) {
+
+    return Modal.open({
+      templateUrl: 'partials/journal/modals/journalErrorViewer.modal.html',
+      controller:  'JournalErrorViewerModalController as JournalErrorViewerModalCtrl',
+      size : 'lg',
+      resolve : {
+        params : function getParams() { return {errors : errors, feedBack : feedBack, cssClass : cssClass}; }
+      }
+    });
+
+  }
+
+  function  getCSSClass (feedBack) {
+    return feedBack.hasError ? 'error' : feedBack.hasWarning ? 'warning' : 'success';
+  }
+
+  function getFeedBack(errors) {
+    var feedBack = {};
+
+    feedBack.hasError = errors.some(function (error) {
+      if(error){
+        return error.fatal;
+      }else{
+        return false;
+      } 
+    });
+    
+    if(feedBack.hasError){
+      feedBack.hasWarning = false;
+      feedBack.hasSuccess = false;      
+      return feedBack;
+    }
+
+    feedBack.hasWarning = errors.some(function (error) {
+      return error;
+    });
+
+    if(feedBack.hasWarning){
+      feedBack.hasSuccess = false;
+      return feedBack;
+    }
+    
+    feedBack.hasSuccess = true;
+    
+    return feedBack;
+  }
+
+  function parseSelectedGridRecord (records){
+    var parsed = [], processedTransactions = [];
+
+    records.forEach(function (record){
+
+      if(processedTransactions.indexOf(record.entity.trans_id) === -1){
+        parsed = parsed.concat(record.treeNode.parentRow.treeNode.children.map(function (child){
+          return child.row.entity;
+        }));
+
+        processedTransactions.push(record.entity.trans_id);
+      }
+    });
+
+    return parsed;
+  }
+
+  /**
+   * @function getTransactionList
+   * @description
+   * take an array of posting journal lines and return an
+   * array of distinct trans_id
+   *
+   * So here is the format of data which can be sent to
+   * this function :
+   *
+   * [{trans_id : xx, ...}, {trans_id : yy, ...}]
+   **/
+  function getTransactionList (lines) {
```

Comment:
FYI, this could also be written like this:
```js
function getTransactionList(lines) {
  return lines.reduce(function (transactions, row) {
    var transId = row.trans_id;
    return (transactions.indexOf(transId) === -1) ? transactions.concat(transId) : transactions;
  }, []);
}
```